tfx/experimental/pipeline_testing/examples/chicago_taxi_pipeline/taxi_pipeline_dummy_e2e_test.py

```diff
@@ -0,0 +1,164 @@
+    self._metadata_path = os.path.join(self._test_dir, 'tfx', 'metadata',
+                                       self._pipeline_name, 'metadata.db')
+
+  def assertExecutedOnce(self, component: Text) -> None:
+    """Check the component is executed exactly once.
+    Pipeline root is <test_dir>/tfx/pipelines/beam_test/
+    Execution outputs are saved in <component.id>/<key>/<execution.id>
+    """
+    component_path = os.path.join(self._pipeline_root, component)
+    self.assertTrue(tf.io.gfile.exists(component_path))
+    outputs = tf.io.gfile.listdir(component_path)
+    for output in outputs:
+      current_dir = os.path.join(component_path, output)
+      absl.logging.info("assertExecutedOnce: checking %s", current_dir)
+      execution = tf.io.gfile.listdir(current_dir)
+      self.assertLen(execution, 1)
+
+  def assertPipelineExecution(self) -> None:
+    self.assertExecutedOnce('CsvExampleGen')
+    self.assertExecutedOnce('Evaluator')
+    self.assertExecutedOnce('ExampleValidator')
+    self.assertExecutedOnce('Pusher')
+    self.assertExecutedOnce('SchemaGen')
+    self.assertExecutedOnce('StatisticsGen')
+    self.assertExecutedOnce('Trainer')
+    self.assertExecutedOnce('Transform')
+
+  def testTaxiPipelineBeam(self):
+    record_dir = os.path.join(os.environ['HOME'],
+                              'tfx/tfx/experimental/pipeline_testing/',
+                              'examples/chicago_taxi_pipeline/testdata')
+
+    component_ids = ['CsvExampleGen', \
+                    'StatisticsGen', 'SchemaGen', \
+                    'ExampleValidator', 'Transform', \
+                    'Trainer', 'Evaluator', 'Pusher']
+
+    MyDummyLauncher = \
+      dummy_component_launcher.create_dummy_launcher_class(record_dir,
+                                                           component_ids,
+                                                           {})
+    taxi_pipeline = taxi_pipeline_beam._create_pipeline(  # pylint:disable=protected-access, unexpected-keyword-arg
+                pipeline_name=self._pipeline_name,
+                data_root=self._data_root,
+                module_file=self._module_file,
+                serving_model_dir=self._serving_model_dir,
+                pipeline_root=self._pipeline_root,
+                metadata_path=self._metadata_path,
+                direct_num_workers=1)
+
+    BeamDagRunner(config=pipeline_config.PipelineConfig(
+        supported_launcher_classes=[
+            MyDummyLauncher,
+        ],
+        )).run(taxi_pipeline)
+
+    self.assertTrue(tf.io.gfile.exists(self._metadata_path))
+    metadata_config = metadata.sqlite_metadata_connection_config(
+        self._metadata_path)
+    with metadata.Metadata(metadata_config) as m:
+      artifact_count = len(m.store.get_artifacts())
+      execution_count = len(m.store.get_executions())
+      self.assertGreaterEqual(artifact_count, execution_count)
+      self.assertEqual(len(taxi_pipeline.components), execution_count)
```

Comment:
nit: self.assertLen() might be better.

Reply:
```
Traceback (most recent call last):
  File "/Users/sujipark/tfx/tfx/experimental/pipeline_testing/examples/chicago_taxi_pipeline/taxi_pipeline_regression_test.py", line 100, in testTaxiPipelineBeam
    self.assertLen(execution_count, len(taxi_pipeline.components))
  File "/Users/sujipark/opt/anaconda3/envs/tfx/lib/python3.7/site-packages/absl/testing/absltest.py", line 823, in assertLen
    '{!r}'.format(type(container).__name__), msg)
  File "/Users/sujipark/opt/anaconda3/envs/tfx/lib/python3.7/site-packages/absl/testing/absltest.py", line 1665, in fail
    return super(TestCase, self).fail(self._formatMessage(prefix, msg))
AssertionError: Expected a Sized object, got: 'int'

```

Reply:
It sould become `self.assertLen(taxi_pipeline.components, execution_count)`.